utils/python/CIME/SystemTests/pet.py

```diff
@@ -40,5 +40,14 @@ def _case_two_setup(self):
         for comp in self._COMPONENT_LIST:
             self._case.set_value("NTHRDS_%s"%comp, 1)
 
+        # The need for this is subtle. On batch systems, the entire PET test runs
+        # under a single submission and that submission is configured based on
+        # the case settings for case 1, IE 2 threads for all components. This causes
+        # the procs-per-node to be half of what it would be for single thread. On some
+        # machines, if the mpiexec tries to exceed the procs-per-node that were given
+        # to the batch submission, things break. Setting MAX_TASKS_PER_NODE to half of
+        # it original value prevents this.
+        self._case.set_value("MAX_TASKS_PER_NODE", self._case.get_value("MAX_TASKS_PER_NODE") / 2)
+
         # Need to redo case_setup because we may have changed the number of threads
         case_setup(self._case, reset=True)
```

Comment:
Sorry to weigh in late on this - I think that the intended solution here is that the case_setup call at line 53 should include the test_mode=True argument.   Do you mind testing again with that change instead of this one?  Thanks

Reply:
Since there is now a new variable `MAX_MPITASKS_PER_NODE`, halving probably needs to be done for `MAX_MPITASKS_PER_NODE` too. For example, if the case-dir starts out with MAX_TASKS_PER_NODE equal to MAX_MPITASKS_PER_NODE, then this division makes MAX_TASKS_PER_NODE (ranks and threads) less than MAX_MPITASKS_PER_NODE (only ranks).

Reply:
@amametjanov Is correct that MAX_MPITASKS_PER_NODE should always be <= MAX_TASKS_PER_NODE.   Note though that this is not a new variable, in the past it was called PES_PER_NODE, it's name was recently changed to clarify its intent.